accelerator/core/serializer/serializer.c

```diff
@@ -729,6 +846,51 @@ status_t ta_send_transfer_res_serialize(ta_send_transfer_res_t* res, char** obj)
   return ret;
 }
 
+status_t send_mam_req_deserialize(const char* const obj, ta_send_mam_req_t* const req) {
+  if (obj == NULL || req == NULL) {
+    ta_log_error("%s\n", "SC_SERIALIZER_NULL");
+    return SC_SERIALIZER_NULL;
+  }
+  status_t ret = SC_OK;
+  cJSON* json_obj = cJSON_Parse(obj);
+
+  const int protocol_len = 20;
+  char protocol_str[protocol_len];
+
+  if (json_obj == NULL) {
+    ret = SC_SERIALIZER_JSON_PARSE;
+    ta_log_error("%s\n", "SC_SERIALIZER_JSON_PARSE");
+    goto done;
+  }
+
+  ret = ta_json_get_string(json_obj, "protocol", protocol_str, protocol_len);
+  if (ret != SC_OK) {
+    ta_log_error("%d\n", ret);
+    goto done;
+  }
+  if (!strncmp(protocol_str, "MAM_V1", strlen("MAM_V1"))) {
+    req->protocol = MAM_V1;
+  }
+
+  switch (req->protocol) {
```

Comment:
I dislike the repeated switch-case every time MAMv1 is addressed. Can you rework?

Reply:
So should I remove the switch-case here at this moment, since we only support MAMv1 at this moment? Or?

Reply:
The support for further MAMv1.1 should be straightforward. I don't think MAMv1.1 would be completely different from MAMv1. Let's simply protocol dispatching.